src-electron/upgrade/upgrade.js

```diff
@@ -39,15 +46,49 @@ const fsp = fs.promises
 async function getForcedExternalStorage(db, attributeId) {
   let pkgs = await queryPackage.getPackageRefByAttributeId(db, attributeId)
   let zcl = await queryPackage.getPackageByPackageId(db, pkgs)
-  zcl = zcl.path
-  let obj = await fsp.readFile(zcl)
-  let data = JSON.parse(obj)
+  zcl = zcl?.path
+  let obj = await fsp.readFile(zcl, 'utf-8')
+  let data = await parseJson(obj)
   let byName = data?.attributeAccessInterfaceAttributes
   let lists = data?.listsUseAttributeAccessInterface
   let forcedExternal = { byName, lists }
   return forcedExternal
 }
 
+/**
+ * This function takes a clusterId (the database ID, not the specification-defined ID) and an array of attributes (associated with the database defined clusterID)
+ * and changes the global attributes (attributes with specification defined clusterId = null) to represent storage policy
+ * based on the cluster/attribute pair in zcl.json
+ *
+ * Although the specification defined clusterID of the attriibute is null indicating it is a global attribute, we know what the database defined clusterID is by what is passed in as a parameter.
+ *
+ * That database defined clusterID is used to query the name of the cluster which is in turn used to compute the storage policy for that cluster/attribute pair.
+ *
+ * @export
+ * @param {*} db
+ * @param {*} clusterId (database defined) the clusterId representing a cluster from the database being used in the application
+ * @param {*} attributes an array of objects representing the attributes associated with the cluster
+ * @returns an array of objects representing attributes in the database
+ */
+
+async function computeStoragePolicy(db, clusterId, attributes) {
```

Comment:
```suggestion
async function computeStoragePolicyForGlobalAttributes(db, clusterId, attributes) {
```
if that's what it does.